lib/core/connector_job.rb

```diff
@@ -62,16 +63,36 @@ def canceling?
       status == Connectors::SyncStatus::CANCELING
     end
 
+    def suspended?
+      status == Connectors::SyncStatus::SUSPENDED
+    end
+
+    def canceled?
+      status == Connectors::SyncStatus::CANCELED
+    end
+
+    def pending?
+      Connectors::SyncStatus::PENDING_STATUSES.include?(status)
+    end
+
+    def active?
+      Connectors::SyncStatus::ACTIVE_STATUSES.include?(status)
+    end
+
+    def terminated?
+      Connectors::SyncStatus::TERMINAL_STATUSES.include?(status)
+    end
+
     def connector_snapshot
-      self[:connector]
+      self[:connector] || {}
     end
 
     def connector_id
       connector_snapshot[:id]
     end
 
     def index_name
-      connector_snapshot[:configuration]
+      connector_snapshot[:index_name]
     end
```

Comment:
I see, so `connector_snapshot` defaults to empty object cause of these lines.

I think it's just making code/debugging harder: the code that tries to get id/index_name for non-existing connector snapshot will return empty values, while in reality it should not have attempted to access connector snapshot id/index name.

I think it's worth to only return `connector_snapshot` from the job and let the user of it handle nulls or return a wrapper class that encapsulates some of its logic.

That is a non-blocking comment though, this refactor can be done later.

Reply:
I added `|| {}` defensively.

The snapshot data (e.g. `configuration`, `pipeline`, etc..) will be used in job execution, and they are expected to be there (the last item of https://github.com/elastic/enterprise-search-team/issues/3283#issue-1442211849). What I will do is something like this:
```ruby
  pipeline = job.pipeline || connector_settings.pipeline
```